### ISSUE TITLE
handle limits of 0 for a log-scale axis?

### ISSUE THREAD

**Issue by lindsayplatt** [USGS-R/gsplot#223]
Having the limit of 0 on a log-scale axis will throw in error because `log(0)=-inf`. Should we provide a work around for this error or just let it happen?

**Comment by lindsayplatt** [comment]
Laura's comments from brief slack discussion:

```
I was going back-and-forth if we'd offer a "0" limit option for the user in gsplot...it's a pain I deal with a lot, but at the same time, they should know that asking for a limit of 0 in log scale is a no-no

but what I do is take 1/10th the minimum, and change the axis to "ND" there (non-detect) instead of whatever it's suppose to be

kind of hacky

sorry, change the zero's to 1/10th the minimum detected levels
```

**Comment by aappling-usgs** [comment]
I like the idea to change the axis to "ND" there, and I agree that sometimes it's nice to be able to use a log scale for data with 0's, despite the infinity thing. I do think you should give a warning if you're getting rid of the error - a user should always know about it when she's plotting log(0) values.

Less sure about always changing 0's to 1/10th of the next-smallest value - specific datasets (e.g., those with a very wide or very compressed range) might end up with a very small or very large gap between the smallest non-0 data and the 0 (ND) data. Could you make 1/10th the default but have an option for the user to pick a different value when needed?

**Comment by ldecicco-USGS** [comment]
Yeah, definitely need a warning or message, and the 1/10th could be an argument (with the default being 0.1). 

It could be even smarter: the default could be the next lower "pretty" log scale marker (that's what I'm lazily going for with 1/10).

**Comment by aappling-usgs** [comment]
I really like the next lower pretty marker idea!

**Comment by ldecicco-USGS** [comment]
Editted:

```
x <- rnorm(100, 50000, 10000)
logScale <- FALSE
if(logScale){
  tick1 <- axisTicks(range(log10(x)), log = TRUE)
  tick2 <- tick1[2:length(tick1)]
  resol <- mean(log10(tick2) - log10(tick1[-length(tick1)]))
  newAT <- c(tick1, 10^(max(log10(tick1))+resol))
  plot(x, ylim=range(newAT), log="y")
} else {
  tick1 <- axisTicks(range(x), log = FALSE)
  tick2 <- tick1[2:length(tick1)]
  resol <- mean(tick2 - tick1[-length(tick1)])
  newAT <- c(tick1, max(tick1)+resol)
  plot(x, ylim=range(newAT))
}

```

was just playing with this a bit for something else. I bet there's a better/simplier way to get an extra 'pretty' tick.
